test/unit/domain/consents/revoke.test.ts

```diff
@@ -0,0 +1,229 @@
+  credentialChallenge: 'xyhdushsoa82w92mzs='
+}
+
+const requestBody: SDKStandardComponents.PatchConsentsRequest = {
+  status: 'REVOKED',
+  revokedAt: 'now'
+
+}
+
+describe('server/domain/consents/revoke', (): void => {
+  beforeAll((): void => {
+    mockConsentUpdate.mockResolvedValue(2)
+    mockLoggerError.mockReturnValue(null)
+    mockLoggerPush.mockReturnValue(null)
+  })
+
+  beforeEach((): void => {
+    jest.clearAllMocks()
+  })
+
+  describe('isConsentRequestInitiatedByValidSource', (): void => {
+    it('Should return true', (): void => {
+      expect(
+        isConsentRequestInitiatedByValidSource(partialConsentActive, request))
+        .toBe(true)
+    })
+
+    it('Should return false because consent is null', (): void => {
+      expect(isConsentRequestInitiatedByValidSource(
+        null as unknown as Consent, request))
+        .toBeFalsy()
+    })
+
+    it('Should return false because initiator ID does not match', (): void => {
+      expect(
+        isConsentRequestInitiatedByValidSource(partialConsentActive2, request))
+        .toBeFalsy()
+    })
+
+    it('Should throw an error as request headers are missing', (): void => {
+      expect((): void => {
+        isConsentRequestInitiatedByValidSource(
+          partialConsentActive, requestNoHeaders as Request)
+      }).toThrowError()
+    })
+  })
+
+  describe('revokeConsentStatus', (): void => {
+    it('Should return a revoked consent', async (): Promise<void> => {
+      const revokedConsent = await revokeConsentStatus(partialConsentActive)
+      expect(revokedConsent.status).toBe('REVOKED')
+      expect(revokedConsent.revokedAt).toBeDefined()
+      expect(mockConsentUpdate).toHaveBeenCalled()
+      expect(mockLoggerPush).not.toHaveBeenCalled()
+    })
+
+    it('Should also return a revoked consent', async (): Promise<void> => {
+      const revokedConsent = await revokeConsentStatus(completeConsentActive)
+      expect(revokedConsent.status).toBe('REVOKED')
+      expect(revokedConsent.revokedAt).toBeDefined()
+      expect(mockConsentUpdate).toHaveBeenCalled()
+      expect(mockLoggerPush).not.toHaveBeenCalled()
+    })
+
+    it('Should return the consent without any operations, if already revoked',
+      async (): Promise<void> => {
+        const revokedConsent = await revokeConsentStatus(completeConsentRevoked)
+        expect(revokedConsent).toStrictEqual(completeConsentRevoked)
+        expect(mockLoggerPush).toHaveBeenCalled()
+        expect(mockConsentUpdate).not.toHaveBeenCalled()
+      })
+
+    it('Should throw an error due to error in updating consent',
+      async (): Promise<void> => {
+        mockConsentUpdate.mockRejectedValue(new Error('Test Error'))
+
+        await expect(revokeConsentStatus(partialConsentActive2))
+          .rejects
+          .toThrowError('Test Error')
+
+        expect(mockConsentUpdate).toHaveBeenCalled()
+        expect(mockLoggerPush).not.toHaveBeenCalled()
+      })
+  })
+
```

Comment:
there is a lack of tests for cases when `revokeConsentStatus` is called with  REVOKED Consent arg.